server/banned.go

```diff
@@ -37,11 +37,12 @@ func (pi *packetInfo) checkBanMessage() error {
 		}
 
 		bannedTrackerData.mutex.Lock()
-		defer bannedTrackerData.mutex.Unlock()
 		bannedTrackerData.bannedBy[td.verificationKey] = nil
+		bannedTrackerData.mutex.Unlock()
```

Comment:
My understanding of our discussion: this immediate unlock creates a race condition for multiple blames where the player could be detected as pool banned multiple times. However it can't be locked for the subsequent code.

Instead, just return if an IP ban is detected here because there is no point in submitting multiple bans at the same time.

Reply:
This is resolved in the latest commit.